### PR TITLE
backend/feat : added verification message for dl and rc

### PR DESCRIPTION
**Pull request by AshishSaini25**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

added verification message for dl and rc

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Review by piyushKumar-1 — DISMISSED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/DriverLicense.hs [hunk 1]
```diff
@@ -124,6 +124,8 @@ verifyDL isDashboard mbMerchant (personId, _, merchantOpCityId) req@DriverDLReq
     Just driverLicense -> do
       unless (driverLicense.driverId == personId) $ throwImageError imageId1 DLAlreadyLinked
       unless (driverLicense.licenseExpiry > now) $ throwImageError imageId1 DLAlreadyUpdated
+      let rules = listToMaybe driverLicense.failedRules
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/DriverLicense.hs:127]: Just add one more check if `driverLicense` is `INVALID` then throw error contact support. You driver license class of vehicles are not supported. If it is `VALID` then don't do anything and say `DLAlreadyUpdated`

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/DriverLicense.hs [hunk 2]
```diff
@@ -232,6 +234,9 @@ createDL ::
 createDL configs driverId output id imageId1 imageId2 now edl expiry = do
   let classOfVehicles = maybe [] (map (.cov)) output.cov_details
   let verificationStatus = validateDLStatus configs expiry classOfVehicles now
+  let failedRules = case output.status of
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/DriverLicense.hs:237]: No need to put anything in failedRules.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/Status.hs [hunk 3]
```diff
@@ -85,48 +87,75 @@ getAadhaarStatus personId = do
...
     Nothing -> return (NO_DOC_AVAILABLE, Nothing)
 
-getDLAndStatus :: Id SP.Person -> Id DMOC.MerchantOperatingCity -> Int -> Flow (ResponseStatus, Maybe DL.DriverLicense)
+getDLAndStatus :: Id SP.Person -> Id DMOC.MerchantOperatingCity -> Int -> Flow (ResponseStatus, Maybe DL.DriverLicense, Maybe Text)
 getDLAndStatus driverId merchantOpCityId onboardingTryLimit = do
   mDriverLicense <- DLQuery.findByDriverId driverId
-  status <-
+  (status, message) <-
     case mDriverLicense of
-      Just driverLicense -> return $ mapStatus driverLicense.verificationStatus
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/Status.hs:93]: If DL or RC is there then throw invalid vehicle class error.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/Status.hs [hunk 4]
```diff
@@ -85,48 +87,75 @@ getAadhaarStatus personId = do
...
+rcVerificationMessage :: ResponseStatus -> RC.VehicleRegistrationCertificate -> Maybe Text
+rcVerificationMessage status rc = do
+  if status == INVALID || status == FAILED
+    then do
+      let msg = listToMaybe rc.failedRules
+      case msg of
+        Just "id_not_found" -> Just "Wrong RC number entered,Please enter a valid RC number and try again"
+        _ -> Just "Try again after sometime"
+    else Nothing
+
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/Status.hs:158]: In `verificationStatus` function, at `Just req` case, add one another case match on `req.status`. If req.status contains `id_not_found` id not found error message and similarly for source_down etc.

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by hkmangla — DISMISSED** [review]

**Review by khuzema786 — APPROVED** [review]

**Pull request merged**